daemon/src/main/java/org/lsposed/lspd/service/CLIValidator.java

```diff
@@ -0,0 +1,82 @@
+package org.lsposed.lspd.service;
+
+import org.lsposed.lspd.util.SignInfo;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CLIValidator {
+
+    private static final Set<Integer> pidSet = ConcurrentHashMap.newKeySet();
+
+    private static String readProcessCmdLine(int pid) throws IOException {
+        var inputStream = new FileInputStream("/proc/" + pid + "/cmdline");
+        var cmdArr = new BufferedReader(new InputStreamReader(inputStream)).readLine().split("\u0000");
+        return String.join(" ", cmdArr);
+    }
+
+    private static String getDaemonPath() throws IOException {
+        var inputStream = Runtime.getRuntime().exec("magisk --path").getInputStream();
+        var modulesRoot = new BufferedReader(new InputStreamReader(inputStream)).readLine();
+        return modulesRoot + "/.magisk/modules/" + ConfigManager.getInstance().getApi().toLowerCase() + "_lsposed/daemon.apk";
+    }
+
+    // Not safe, this method is easily bypassed
+    private static boolean validProcessName(int pid) throws IOException {
+        var processCmdline = readProcessCmdLine(pid);
+        var daemonPath = getDaemonPath();
+        var cmdline = "app_process -Djava.class.path=" + daemonPath + " /system/bin org.lsposed.lspd.cli.Main";
```

Comment:
check /proc/pid/exe's indoe

Reply:
/proc/pid/exe is a link to app_process, maybe we can remove this check, when the pin code authorization will be implemented.

Reply:
True.

Reply:
with pin code this check has been removed